homeassistant/components/magichome/switch.py

```diff
@@ -0,0 +1,41 @@
+"""Platform for MagicHome switches integration."""
+from homeassistant.components.switch import ENTITY_ID_FORMAT, SwitchDevice
+
+from . import MAGICHOME_API, MagicHomeDevice
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up MagicHome Switch device."""
+    if discovery_info is None:
+        return
+    magichome = hass.data[MAGICHOME_API]
+    dev_ids = discovery_info.get("dev_ids")
+    devices = []
+    for dev_id in dev_ids:
+        device = magichome.get_device_by_id(dev_id)
+        if device is None:
+            continue
+        devices.append(MagicHomeSwitch(device))
+    add_entities(devices)
+
+
+class MagicHomeSwitch(MagicHomeDevice, SwitchDevice):
```

Comment:
I know @MartinHjelmare commented to remove this file.  When it is added back please note that SwitchDevice is deprecated the new class name is SwitchEntity